test/hotspot/jtreg/gc/shenandoah/options/TestSelectiveBarrierFlags.java

```diff
@@ -26,10 +26,21 @@
  * @summary Test selective barrier enabling works, by aggressively compiling HelloWorld with combinations
  *          of barrier flags
  * @requires vm.gc.Shenandoah
+ * @requires vm.debug
+ * @library /test/lib
+ * @run driver TestSelectiveBarrierFlags -Xint
+ * @run driver TestSelectiveBarrierFlags -Xbatch -XX:CompileThreshold=100 -XX:TieredStopAtLevel=1
+ * @run driver TestSelectiveBarrierFlags -Xbatch -XX:CompileThreshold=100 -XX:-TieredCompilation -XX:+ShenandoahVerifyOptoBarriers
+ */
+
+/* @test
+ * @summary Test selective barrier enabling works, by aggressively compiling HelloWorld with combinations
+ *          of barrier flags
+ * @requires vm.gc.Shenandoah
+ * @requires !vm.debug
```

Comment:
Seems like it might be a bit simpler to have one `@test` with the first two `@run` lines and no `@requires` on `vm.debug`, and a second (debug-only) `@test` with the third `@run`.